File: function.R
synergize <- function(data_frame,
                      el1 = 1,
                      el2 = 3,
                      char1 = 2,
                      char2 = 4,
                      x = "Elements",
                      y = "Characteristics",
                      z = "Total") {
   #selecting elements characteristics inside a list
   element1 <- strsplit(data_frame[, char1], split = ",")
   element2 <- strsplit(data_frame[, char2], split = ",")
   #setting elements names
   element1 <- setNames(element1, as.vector(data_frame[, el1]))
   element2 <- setNames(element2, as.vector(data_frame[, el2]))
   #removing elements without characteristics
   element1 <- element1[sapply(element1, length) != 0]
   element2 <- element2[sapply(element2, length) != 0]
   #setting a list to store loop result
   result <- list()
   #if the element is repeated in both columns, the characteristics will be selected without repetition
   for (i in names(element2))
      for (j in names(element1))
       {if (i == j) {
         output <- union(element2[[i]], element1[[j]])
         result[[i]] <- output
         }
      }
   #setting object back to data frame
   result_df <- list2DF(result)
   #checking if there is some similar element between columns
   if (dim(result_df)[2] > 0)
   #selecting elements names
   {el_id <- vector()
   for (k in colnames(result_df))
   {output <- k
   el_id[k] <- output
   }
   #uniting characteristisc in a string
   char_union <- vector()
   for (l in seq_len(ncol(result_df)))
   {output <- paste(unique(result_df[, l]), collapse = ",")
   char_union[l] <- output
   }
   #counting characteristics
   numb_total <- vector()
   for (m in seq_len(ncol(result_df)))
   {output <- length(unique(result_df[, m]))
   numb_total[m] <- output
   }
   #naming columns and returning value
   table <- data.frame(x = el_id, y = char_union, z = numb_total)
   colnames(table) <- c(x, y, z)
   return(table)
   #if there are no similarities
   } else {
      message("There are no common elements")
      return(NULL)
   }
   }

File: function.py
import pandas as pd

def synergize(data_frame, el1=0, el2=2, char1=1, char2=3, x="Elements", y="Characteristics", z="Total"):
    # selecting elements characteristics inside a pandas.Series
    element1 = data_frame.iloc[:, char1].str.split(',')
    element2 = data_frame.iloc[:, char2].str.split(',')
    # setting elements names and changing to dictionary
    element1 = dict(zip(data_frame.iloc[:, el1], element1))
    element2 = dict(zip(data_frame.iloc[:, el2], element2))
    # removing elements without characteristics, but nan values will cause error
    element1 = {k: v for k, v in element1.items() if len(v) != 1 or v[0] != ''}
    element2 = {k: v for k, v in element2.items() if len(v) != 1 or v[0] != ''}
    # setting a list to store loop result
    result = {}
    # if the element is repeated in both columns, the characteristics will be selected without repetition
    for i in element2:
        for j in element1:
            if i == j:
                output = list(set(element2[i]) | set(element1[j]))
                result[i] = output
    # setting object back to data frame
    result_df = pd.DataFrame(result.items(), columns=[x, y])
    # checking if there is some similar element between columns
    if len(result_df) > 0:
        # uniting characteristisc in a string
        result_df[y] = result_df[y].apply(lambda x: ','.join(list(set(x))))
        # counting characteristics
        result_df[z] = result_df[y].apply(lambda x: len(x.split(',')))
        # naming columns and returning value
        result_df = result_df[[x, y, z]].rename(columns={x: x, y: y, z: z})
        return result_df
    # if there are no similarities
    else:
        print("There are no common elements")
        return None